packages/devextreme/js/ui/date_box/ui.date_box.base.js

```diff
@@ -161,7 +164,16 @@ const DateBox = DropDownEditor.inherit({
             pickerType = PICKER_TYPE.list;
         }
 
-        this.option('showDropDownButton', devices.real().platform !== 'generic' || pickerType !== PICKER_TYPE['native']);
+        const isMozillaOnAndroid = platform === 'android' && browser.mozilla;
+        const isNativePickerType = pickerType === PICKER_TYPE.native;
```

Comment:
Let's extract this logic to separate function.

Reply:
Also let's use _isNativeType() method instead

Reply:
fixed